### ISSUE TITLE
Wrong matched route path in the context

### ISSUE THREAD

**Issue by drsect0r** [gin-gonic/gin#1917]
- go version go1.12 linux/amd64
- gin commit 6e320c97e83a61df3daa0f28695a736bece3104b
- Ubuntu inside Windows Subsystem for Linux

## Description
We were pretty excited to perform code cleanup after #1826 was merged into master. We removed the git checkout and the #1390 patch present in our CI pipeline and worked based upon the latest master. 

We noticed that one endpoint stopped working as expected and discovered that it was being misreported as a different route. As a test, we removed the erroneous route, but the endpoint in question still reported a wrong route path. 

For now, we have restored the #1390 patch on top of commit e5261480fde106ac5385f00fb00f9bc6b4035485 and this resolved the issue.

**Comment by thinkerou** [comment]
@drsect0r can you post mini example in order to reproduce the issue? thanks!

**Comment by bbiao** [comment]
@drsect0r I hava fix it in https://github.com/gin-gonic/gin/issues/1919 you can have a try

**Comment by caihangui** [comment]
I think I had met the same problem

this is my code
```
func main() {
	r := gin.Default()
	r.GET("/ping", func(c *gin.Context) {

		fmt.Println("ping function 's c.FullPath() == ",c.FullPath())

		fmt.Println("c.FullPath() == \"/ping\" : ",c.FullPath() == "/ping")

		c.JSON(http.StatusOK, gin.H{})
	})

	r.GET("/pong", func(c *gin.Context) {

		fmt.Println("pong function 's c.FullPath() == ",c.FullPath())

		fmt.Println("c.FullPath() == \"/pong\" : ",c.FullPath() == "/pong")

		c.JSON(http.StatusOK, gin.H{})
	})

	r.Run()
}
```

but its result don't be expected
```
pong function 's c.FullPath() ==  /pong
c.FullPath() == "/pong" :  true
[GIN] 2019/06/14 - 14:52:58 | 200 |     199.792µs |             ::1 | GET      /pong
ping function 's c.FullPath() ==  /pong
c.FullPath() == "/ping" :  false
[GIN] 2019/06/14 - 14:53:07 | 200 |      42.793µs |             ::1 | GET      /ping
```

**Comment by bbiao** [comment]
My PR can fix your problems 

=============
@caihangui  the output will be like below if you apply my patch

ping function 's c.FullPath() ==  /ping
c.FullPath() == "/ping" :  true
[GIN] 2019/06/14 - 15:23:21 | 200 |       95.29µs |       127.0.0.1 | GET      /ping
pong function 's c.FullPath() ==  /pong
c.FullPath() == "/pong" :  true
[GIN] 2019/06/14 - 15:23:27 | 200 |      44.114µs |       127.0.0.1 | GET      /pong

https://github.com/gin-gonic/gin/pull/1919

**Comment by bbiao** [comment]
@drsect0r you can try this patch https://github.com/gin-gonic/gin/pull/1919

**Comment by drsect0r** [comment]
@bbiao Thank you for your patch! Will apply it tomorrow and report back.

**Comment by thinkerou** [comment]
merged!
